sway/ipc-i3.c

```diff
@@ -25,7 +22,11 @@ static void ipc_i3_json_describe_view(struct sway_container *c, json_object *obj
 	}
 #endif
 
+#ifdef HAVE_XWAYLAND
```

Comment:
I left the fixup commit as-is for review because that really didn't feel like it improved code quality here.

Should I drop this and keep the void casts, or does anyone have a better alternative?

Reply:
GitHub seems to have choked on the rebase, here is the [actual fixup commit](https://github.com/swaywm/sway/pull/2986/commits/466871a89b1ed8067c953be3d944ea9c4491f64b).

Reply:
I'm going to drop this commit and just move the void casts in the `#else` of the HAVE_XWAYLAND check, if that's alright with everyone. That ought to be cleaner.

Reply:
I still don't get why this is needed. If this file is built without Xwayland, then the code becomes:

```c
uint32_t window_id = 0;
…
json_object_object_add(object, "window", json_object_new_int(window_id));
```

Why doesn't this work?

Reply:
Oh, yeah, now that you mention it, the void cast for window_id was an oversight. The only guards needed are for role and parent_id, because the compiler complains that they are effectively unused when compiling without Xwayland. I'll remove the bit about window_id though.

Reply:
Then just add a guard to the declaration of those two?

Reply:
I've updated the fixup commit without the window_id guard, and I moved parent_id and role into the first guard. Is that more in line with what you had in mind?